integration/cluster_test.go

```diff
@@ -416,3 +419,77 @@ func clusterMustProgress(t *testing.T, membs []*member) {
 		mcancel()
 	}
 }
+
+func TestTransferLeader(t *testing.T) {
```

Comment:
I will try what Xiang suggested (except 3) to address `continual put case is what would be expected`. Thanks!